### PR TITLE
User Profile With Google API

### PR DESCRIPTION
**Pull request by upadhyaydhiren**
In this version owner,driver truck related all  user profile activity is completed.
Goods Part is still remaining.
Also I had change Design Pattern of dao and service classes.

### CODE REVIEW

**Comment by upadhyaydhiren** [comment]
Please Update Your repository.

File: nb-configuration.xml [hunk 1]
```diff
@@ -12,13 +12,14 @@ Properties that influence various parts of the IDE, especially code formatting a
...
+-->
+        <org-netbeans-modules-css-prep.less_2e_mappings>/less:/css</org-netbeans-modules-css-prep.less_2e_mappings>
+        <org-netbeans-modules-css-prep.less_2e_enabled>false</org-netbeans-modules-css-prep.less_2e_enabled>
+        <org-netbeans-modules-css-prep.sass_2e_enabled>false</org-netbeans-modules-css-prep.sass_2e_enabled>
+        <org-netbeans-modules-css-prep.sass_2e_compiler_2e_options/>
+        <org-netbeans-modules-css-prep.less_2e_compiler_2e_options/>
+        <org-netbeans-modules-css-prep.sass_2e_mappings>/scss:/css</org-netbeans-modules-css-prep.sass_2e_mappings>
+        <org-netbeans-modules-maven-j2ee.netbeans_2e_hint_2e_deploy_2e_server>gfv3ee6</org-netbeans-modules-maven-j2ee.netbeans_2e_hint_2e_deploy_2e_server>
+        <org-netbeans-modules-javascript2-requirejs.enabled>true</org-netbeans-modules-javascript2-requirejs.enabled>
+    </properties>
```
Comment (shayaneumar) [inline nb-configuration.xml:24]: Should not we git ignore this file?

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 2]
```diff
@@ -71,10 +100,21 @@ public ModelAndView save(@ModelAttribute Owner owner, ModelMap model, BindingRes
             owner.setCreatedBy(owner.getEmail());
             owner.setCreatedDate(new Date());
             owner.setModifiedDate(new Date());
-            if (ownerService.insertOwner(owner)) {
+            UserRole userRole = new UserRole();
+            Role role = new Role();
+            role.setRoleId(1);
+            userRole.setRole(role);
+            userRole.setUser(owner);
+            List<UserRole> roles = new ArrayList<>();
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java]: rename roles  to userroles

File: src/main/resources/DB.properties [hunk 3]
```diff
@@ -4,4 +4,4 @@
 db.classDriver=com.mysql.jdbc.Driver
 db.jdbcurl=jdbc:mysql://localhost:3306/logistics
 db.username=root
-db.password=ecotruck
+db.password=test
```
Comment (shayaneumar) [inline src/main/resources/DB.properties]: Don't inculde this file in commit

File: src/main/webapp/WEB-INF/jsp/profile.jsp [hunk 4]
```diff
@@ -796,10 +789,219 @@
...
+                            else if (name === 'country')
+                            {
+                                $('#country').val(object.long_name);
+                                return false;
+                            }
+                            else if (name === 'postal_code')
+                            {
+                                $('#pincode').val(object.short_name);
+                                return false;
+                            }
```
Comment (shayaneumar) [inline src/main/webapp/WEB-INF/jsp/profile.jsp:960]: Switch case is better option in this case.

File: src/main/webapp/WEB-INF/jsp/profile.jsp [hunk 5]
```diff
@@ -796,10 +789,219 @@
...
+                            else if (name === 'country')
+                            {
+                                $('#countrydriver').val(object.long_name);
+                                return false;
+                            }
+                            else if (name === 'postal_code')
+                            {
+                                $('#pincodedriver').val(object.short_name);
+                                return false;
+                            }
```
Comment (shayaneumar) [inline src/main/webapp/WEB-INF/jsp/profile.jsp:990]: Switch case is better option in this case.

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 6]
```diff
@@ -142,21 +188,15 @@ public ModelAndView completeProfile(@ModelAttribute Owner owner, BindingResult r
                 fetchedOwner.setNoOfVehicles(owner.getNoOfVehicles());
                 fetchedOwner.setBusinessType(owner.getBusinessType());
                 fetchedOwner.setUserReferanceCode(owner.getUserReferanceCode());
-                owner.getAddress().setPincode(Integer.parseInt(request.getParameter("pincode111")));
+                owner.getAddress().setUser(fetchedOwner);
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:325]: Looks likec cyclic reference that might be uncessary.

owner has an address
An address belongs to an owner

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 7]
```diff
@@ -166,10 +206,10 @@ public ModelAndView completeProfile(@ModelAttribute Owner owner, BindingResult r
             return new ModelAndView("registration", map);
         }
     }
-
+    
     @RequestMapping(value = "addtruck", method = RequestMethod.POST)
-    public String getProfile(@ModelAttribute Vehicle vehicle) {
-
+    public String getProfile(@ModelAttribute Vehicle vehicle, HttpServletRequest request) {
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java]: Why is request passed I don't see it is used anywhere.

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 8]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+                return "redirect:home";
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "redirect:home";
+        }
+    }
+    
+    @RequestMapping(value = "adddriver", method = RequestMethod.POST)
+    public String addDriver(@ModelAttribute Driver driver, HttpServletRequest request) {
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java]: remove request if not needed

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 9]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+        try {
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            if ((auth instanceof AnonymousAuthenticationToken)) {
+                return "redirect:login";
+            } else {
+                UserDetails userDetails = (UserDetails) auth.getPrincipal();
+                Owner owner = ownerService.getOwnerByUsername(userDetails.getUsername());
+                driver.setIsActive(Boolean.TRUE);
+                driver.setIsDeleted(Boolean.FALSE);
+                driver.setPassword(new BCryptPasswordEncoder().encode((driver.getFirstName() + "123")));
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:375]: Why is this hardcoding?

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 10]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+                return "redirect:home";
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "redirect:home";
+        }
+    }
+    
+    @RequestMapping(value = "updatepersonal", method = RequestMethod.POST)
+    public String updatePersonal(@ModelAttribute Owner owner, HttpServletRequest request) {
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java]: request, remove if not needed

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 11]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+            } else {
+                UserDetails userDetails = (UserDetails) auth.getPrincipal();
+                Owner fetchedOwner = ownerService.getOwnerByUsername(userDetails.getUsername());
+                Vehicle fetchedVehicle = null;
+                int i = 0;
+                for (; i < fetchedOwner.getVehicles().size(); i++) {
+                    if (fetchedOwner.getVehicles().get(i).getRegNo().equals(vehicle.getRegNo())) {
+                        fetchedVehicle = fetchedOwner.getVehicles().get(i);
+                        break;
+                    }
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:470]: Is there a better way to do this in Java simialr to LINQ in C#

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 12]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+                UserDetails userDetails = (UserDetails) auth.getPrincipal();
+                Owner fetchedOwner = ownerService.getOwnerByUsername(userDetails.getUsername());
+                Driver fetchedDriver = null;
+                int i = 0;
+                for (; i < fetchedOwner.getDriverlist().size(); i++) {
+                    if (Objects.equals(fetchedOwner.getDriverlist().get(i).getId(), driver.getId())) {
+                        fetchedDriver = fetchedOwner.getDriverlist().get(i);
+                        break;
+                    }
+                }
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:512]: Is there a better way to do this in Java simialr to LINQ in C#

File: src/main/java/com/utilaider/logistics/domain/VehiclePermits.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+package com.utilaider.logistics.domain;
+
+/**
+ *
+ * @author Dhiren
+ */
+public enum VehiclePermits {
+
+    ALL_INDIA_PERMITS("ALL INDIA PERMITS", "aip"),
+    STATE_PERMITS("STATE_PERMITS", "sip");
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/domain/VehiclePermits.java]: sip or sp?

File: src/main/java/com/utilaider/logistics/domain/VehiclePermits.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package com.utilaider.logistics.domain;
+
+/**
+ *
+ * @author Dhiren
+ */
+public enum VehiclePermits {
+
+    ALL_INDIA_PERMITS("ALL INDIA PERMITS", "aip"),
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/domain/VehiclePermits.java:14]: What are these construct? I am not familiar to this

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 15]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+                return "redirect:login";
+            } else {
+                UserDetails userDetails = (UserDetails) auth.getPrincipal();
+                Owner fetchedOwner = ownerService.getOwnerByUsername(userDetails.getUsername());
+                Vehicle fetchedVehicle = null;
+                int i = 0;
+                for (; i < fetchedOwner.getVehicles().size(); i++) {
+                    if (fetchedOwner.getVehicles().get(i).getRegNo().equals(regno)) {
+                        fetchedVehicle = fetchedOwner.getVehicles().get(i);
+                        break;
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:560]: Is there a better way to do this in Java simialr to LINQ in C#

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java [hunk 16]
```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
...
+            if ((auth instanceof AnonymousAuthenticationToken)) {
+                return "redirect:login";
+            } else {
+                UserDetails userDetails = (UserDetails) auth.getPrincipal();
+                Owner fetchedOwner = ownerService.getOwnerByUsername(userDetails.getUsername());
+                Driver fetchedDriver = null;
+                int i = 0;
+                for (; i < fetchedOwner.getDriverlist().size(); i++) {
+                    if (fetchedOwner.getDriverlist().get(i).getMobile().equals(mobile)) {
+                        fetchedDriver = fetchedOwner.getDriverlist().get(i);
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:601]: Is there a better way to do this in Java simialr to LINQ in C#

File: src/main/java/com/utilaider/logistics/dao/GenericDao.java [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ *
+ * @author Dhiren
+ * @param <PK>
+ * @param <T>
+ */
+public interface GenericDao<PK, T> {
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/dao/GenericDao.java:17]: :100:

File: src/main/java/com/utilaider/logistics/dao/GenericDaoImpl.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ *
+ * @author Dhiren
+ * @param <PK>
+ * @param <T>
+ */
+@Repository
+@Transactional
+public abstract class GenericDaoImpl<PK extends Serializable, T> implements GenericDao<PK, T> {
```
Comment (shayaneumar) [inline src/main/java/com/utilaider/logistics/dao/GenericDaoImpl.java:25]: :100:

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java (hunk 2 above)
Comment (upadhyaydhiren) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java]: I will do this change

File: src/main/resources/DB.properties (hunk 3 above)
Comment (upadhyaydhiren) [inline src/main/resources/DB.properties]: ok
Next Time I will discard this file in commit

File: src/main/webapp/WEB-INF/jsp/profile.jsp (hunk 4 above)
Comment (upadhyaydhiren) [inline src/main/webapp/WEB-INF/jsp/profile.jsp:960]: I will convert into switch

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java (hunk 6 above)
Comment (upadhyaydhiren) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:325]: This is bi-directional relationship
I will use shared primary key.
In Address class User class use as primary key

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java (hunk 7 above)
Comment (upadhyaydhiren) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java]: I will remove

File: src/main/java/com/utilaider/logistics/controller/LogisticsController.java (hunk 9 above)
Comment (upadhyaydhiren) [inline src/main/java/com/utilaider/logistics/controller/LogisticsController.java:375]: Currently We have no feature to driver add own-self in system.
So I had use first-name with 123.

File: src/main/java/com/utilaider/logistics/domain/VehiclePermits.java (hunk 13 above)
Comment (upadhyaydhiren) [inline src/main/java/com/utilaider/logistics/domain/VehiclePermits.java]: BY Mistake.
I will correct

File: nb-configuration.xml (hunk 1 above)
Comment (upadhyaydhiren) [inline nb-configuration.xml:24]: This is ide generated file.
I am not sure.if we add it in git ignore may be impact.

**Pull request closed without merging**